Makefile

```diff
@@ -1,15 +1,144 @@
-dbt-docs:
-	@echo "Generating docs and spinning up the a webserver on port 8081..."
-	@docker-compose run -p "8081:8081" dbt_image bash -c "dbt deps && dbt docs generate -t prod && dbt docs serve --port 8081"
+# ====================================================================================
+# Variables
 
-generate-dbt-docs:
-	@echo "Generating docs"
-	@docker-compose run dbt_image bash -c "dbt deps && dbt docs generate -t prod"
+## General Variables
+# Branch Variables
+PROTECTED_BRANCH := master
+CURRENT_BRANCH   := $(shell git rev-parse --abbrev-ref HEAD)
+# Use repository name as application name
+APP_NAME    := $(shell basename -s .git `git config --get remote.origin.url`)
+# Get current commit
+APP_COMMIT  := $(shell git log --pretty=format:'%h' -n 1)
+# Check if we are in protected branch, if yes use `protected_branch_name-sha` as app version.
+# Else check if we are in a release tag, if yes use the tag as app version, else use `dev-sha` as app version.
+APP_VERSION ?= $(shell if [ $(PROTECTED_BRANCH) = $(CURRENT_BRANCH) ]; then echo $(PROTECTED_BRANCH); else (git describe --abbrev=0 --exact-match --tags 2>/dev/null || echo dev-$(APP_COMMIT)) ; fi)
 
-data-image:
-	@echo "Attaching to data-image and mounting repo..."
-	@docker-compose run data_image bash
+# Get current date and format like: 2022-04-27 11:32
+BUILD_DATE  := $(shell date +%Y-%m-%d\ %H:%M)
 
-dbt-bash:
-	@echo "Running bash with dbt..."
-	@docker-compose run dbt_image bash -c "dbt deps && /bin/bash"
\ No newline at end of file
+## General Configuration Variables
+# We don't need make's built-in rules.
+MAKEFLAGS     += --no-builtin-rules
+# Be pedantic about undefined variables.
+MAKEFLAGS     += --warn-undefined-variables
+# Set help as default target
+.DEFAULT_GOAL := help
+
+## Docker Variables
+
+# Docker executable
+DOCKER                  := $(shell which docker)
+# Dockerfile's location
+DOCKER_FILE             += ./build/Dockerfile
+# Docker compose DBT file
+DOCKER_COMPOSE_DBT_FILE += ./build/docker-compose.dbt.yml
+
+## Python Variables
+# Python executable
+PYTHON                       := $(shell which python)
+# Poetry executable
+POETRY                       := $(shell which poetry)
+# Extract Python Version
+PYTHON_VERSION               ?= $(shell ${PYTHON} --version | cut -d ' ' -f 2 )
+# Temporary folder to output generated artifacts
+PYTHON_OUT_BIN_DIR           := ./dist
+
+# Docker compose specific environment configuration file
+DOCKER_COMPOSE_ENV_FILE      := .dbt.env
+
+## Github Variables
+# A github access token that provides access to upload artifacts under releases
+GITHUB_TOKEN                 ?= a_token
+# Github organization
+GITHUB_ORG                   := mattermost
+# Most probably the name of the repo
+GITHUB_REPO                  := ${APP_NAME}
```

Comment:
Most of the variables here are unused. 
Let's clear them up to improve readability.